### DISCUSSION TITLE
How to consume a queue after it is sharded?

### DISCUSSION THREAD

**Discussion by springmvcGitHub** [rabbitmq/rabbitmq-server#6435] [Other]
hi rabbitmq expert. 
Here is a question. The scenario is as follows: I deployed rabbitmq with K8S, 3 pods, and used rabbitmq_sharding to sharded. The sharding configuration is as follows: '{"shards per node": 1, "routing key": "#"}. The sharding results are shown in the following image:
<img width="1387" alt="image" src="https://user-images.githubusercontent.com/42227896/202662077-c77ce5ab-3a17-4892-8213-4a7030cae969.png">

The problem is: if i use 'history' to consume queue messages, i cannot receive messages (if i use 1 rabbitmq pod to shard multiple pieces, it can use this queue name to consume), but i use 
**sharding: history- rabbit@my-rabbitmq-0.my-rabbitmq-headless.default.svc.cluster.local -0** or 
**sharding: history- rabbit@my-rabbitmq-1.my-rabbitmq-headless.default.svc.cluster.local -0** or 
**sharding: history- rabbit@my-rabbitmq-2.my-rabbitmq-headless.default.svc.cluster.local -0** can receive messages from the queue. How can I use queue name is 'history' to receive messages from all shard queues?

thanks a lot!

_Originally posted by @springmvcGitHub in https://github.com/rabbitmq/rabbitmq-server/discussions/6425#discussioncomment-4175205_

**Comment by michaelklishin** [comment]
From the [`rabbitmq_sharding` README section on consumers](https://github.com/rabbitmq/rabbitmq-server):

```
if you have a shard called images, then you can directly consume from a queue called images.
```

in other words, you do not consume from individual shards. They are an implementation detail.

**Reply by springmvcGitHub** [reply]
> hi rabbitmq expert. Here is a question. The scenario is as follows: I deployed rabbitmq with K8S, 3 pods, and used rabbitmq_sharding to sharded. The sharding configuration is as follows: '{"shards per node": 1, "routing key": "#"}. The sharding results are shown in the following image: <img alt="image" width="1387" src="https://user-images.githubusercontent.com/42227896/202662077-c77ce5ab-3a17-4892-8213-4a7030cae969.png">
> 
> The problem is: if i use 'history' to consume queue messages, i cannot receive messages (if i use 1 rabbitmq pod to shard multiple pieces, it can use this queue name to consume), but i use **sharding: history- [rabbit@my-rabbitmq-0.my-rabbitmq-headless.default.svc.cluster.local](mailto:rabbit@my-rabbitmq-0.my-rabbitmq-headless.default.svc.cluster.local) -0** or **sharding: history- [rabbit@my-rabbitmq-1.my-rabbitmq-headless.default.svc.cluster.local](mailto:rabbit@my-rabbitmq-1.my-rabbitmq-headless.default.svc.cluster.local) -0** or **sharding: history- [rabbit@my-rabbitmq-2.my-rabbitmq-headless.default.svc.cluster.local](mailto:rabbit@my-rabbitmq-2.my-rabbitmq-headless.default.svc.cluster.local) -0** can receive messages from the queue. How can I use queue name is 'history' to receive messages from all shard queues?
> 
> thanks a lot!
> 
> _Originally posted by @springmvcGitHub in [#6425 (comment)](https://github.com/rabbitmq/rabbitmq-server/discussions/6425#discussioncomment-4175205)_

thanks for reply.
I search a lot of document and did my own trial. If the K8S is used to deploy Rabbitmq, and there is only one pod, and the queue is sharded, it can be consumed using images(queue name). However, if the number of pods is more than one and the queue sharded, the queue name cannot be consumed using images. 
How to solve this problem?
